src/main/java/io/vertx/ext/mail/MailMessage.java

```diff
@@ -0,0 +1,250 @@
+package io.vertx.ext.mail;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+@DataObject
+public class MailMessage {
+
+  private String bounceAddress;
+  private String from;
+  private List<String> to;
+  private List<String> cc;
+  private List<String> bcc;
+  private String subject;
+  private String text;
```

Comment:
an email may contain a text and a html part, this is message/alternative where the client can choose which part to show to have a message fitting the display device (in theory this could even be html/rtf/text), most newsletter services use this to display a non-graphic version of the newsletter or at least an explanation text that the message is only visible in html

https://github.com/vert-x3/vertx-mail-service/blob/initial-work/src/test/java/io/vertx/ext/mail/MailTest.java#L79